File: QadeerApp.Web/Modules/Cv/EmployeeExperience/EmployeeExperienceRow.cs
namespace QadeerApp.Cv;

[ConnectionKey("Default"), Module("Cv"), TableName("EmployeeExperiences")]
[DisplayName("Employee Experiences"), InstanceName("Experience")]
[ReadPermission(CvPermissionKeys.EmployeeExperience.View)]
[ModifyPermission(CvPermissionKeys.EmployeeExperience.Update)]
[InsertPermission(CvPermissionKeys.EmployeeExperience.Insert)]
[UpdatePermission(CvPermissionKeys.EmployeeExperience.Update)]
[DeletePermission(CvPermissionKeys.EmployeeExperience.Delete)]
public sealed class EmployeeExperienceRow : Serenity.Extensions.Entities.LoggingRow<EmployeeExperienceRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Experience Id"), Identity, IdProperty]
    public int? EmployeeExperienceId { get => fields.EmployeeExperienceId[this]; set => fields.EmployeeExperienceId[this] = value; }

    [DisplayName("Employee CV"), NotNull, ForeignKey(typeof(EmployeeCvRow)), LeftJoin("cv"), Updatable(false)]
    public int? EmployeeCvId { get => fields.EmployeeCvId[this]; set => fields.EmployeeCvId[this] = value; }

    [DisplayName("Experience Name"), Size(200), NotNull, QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    [DisplayName("User Id"), Expression("cv.[UserId]")]
    public int? UserId { get => fields.UserId[this]; set => fields.UserId[this] = value; }

    public class RowFields : Serenity.Extensions.Entities.LoggingRowFields
    {
        public Int32Field EmployeeExperienceId;
        public Int32Field EmployeeCvId;
        public StringField Name;
        public Int32Field UserId;
    }
}

File: QadeerApp.Web/Modules/ServerTypes/Cv/EmployeeExperienceRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface EmployeeExperienceRow {
    EmployeeExperienceId?: number;
    EmployeeCvId?: number;
    Name?: string;
    UserId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class EmployeeExperienceRow {
    static readonly idProperty = 'EmployeeExperienceId';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Cv.EmployeeExperience';
    static readonly deletePermission = 'Cv:EmployeeExperience:Delete';
    static readonly insertPermission = 'Cv:EmployeeExperience:Insert';
    static readonly readPermission = 'Cv:EmployeeExperience:View';
    static readonly updatePermission = 'Cv:EmployeeExperience:Update';

    static readonly Fields = fieldsProxy<EmployeeExperienceRow>();
}
